packages/grid/x-data-grid/src/hooks/features/pagination/gridPaginationSelector.ts

```diff
@@ -4,60 +4,73 @@ import {
   gridVisibleSortedRowEntriesSelector,
   gridVisibleSortedRowIdsSelector,
   gridVisibleSortedTopLevelRowEntriesSelector,
+  gridVisibleTopLevelRowCountSelector,
 } from '../filter/gridFilterSelector';
 import { gridRowMaximumTreeDepthSelector, gridRowTreeSelector } from '../rows/gridRowsSelector';
+import { getPageCount } from './gridPaginationUtils';
 
 /**
+ * Get the pagination model
  * @category Pagination
- * @ignore - do not document.
  */
-export const gridPaginationSelector = (state: GridStateCommunity) => state.pagination;
+export const gridPaginationModelSelector = createSelector(
+  (state: GridStateCommunity) => state,
```

Comment:
A workaround for providing [direct access selector](https://mui.com/x/react-data-grid/state/#direct-selector-access), as this selector is likely to be used more often by end users.

We can also think about refactoring `createSelector` to support single selector use-case like this maybe?

Reply:
It's somewhat related to https://github.com/mui/mui-x/pull/7147#discussion_r1069682438